src/recon.rs

```diff
@@ -1562,7 +1551,9 @@ pub(crate) fn rav1d_read_coef_blocks<BD: BitDepth>(
                             let a_len = t_dim.w as usize;
                             let l_start = by4 + y as usize;
                             let l_len = t_dim.h as usize;
-                            let cf_idx = ts.frame_thread[1].cf.get();
+                            let cf = CfSelect::Frame(ts.frame_thread[1].cf.get_update(|i| {
+                                i + cmp::min(t_dim.w, 8) as u32 * cmp::min(t_dim.h, 8) as u32 * 16
+                            }));
```

Comment:
This changes the ordering of the set, why is that ok?

Reply:
This is the pattern used in most other places in C, and when I moved it up, all of the tests still passed so I assumed it was okay.

Reply:
Ok looking through I also don't see any uses of `cf` in the `decode_coefs` call stack. Maybe put a comment in that we changed this and it should not be accessed inside `decode_coefs`.

Reply:
Comment added in [`7065142..6155d6b`](https://github.com/memorysafety/rav1d/compare/706514243a4f422029fa4798e1177c26572cf0b7..6155d6b0915ab8d4cef5692f20e5fbf95b47a645).